packages/compartment-mapper/src/policy.js

```diff
@@ -1,28 +1,89 @@
 // @ts-check
 
 /** @typedef {import('./types.js').PackageNamingKit} PackageNamingKit */
+/** @typedef {import('./types.js').AttenuationDefinition} AttenuationDefinition */
+/** @typedef {import('./types.js').FullAttenuationDefinition} FullAttenuationDefinition */
+/** @typedef {import('./types.js').ImplicitAttenuationDefinition} ImplicitAttenuationDefinition */
+/** @typedef {import('./types.js').Attenuator} Attenuator */
+/** @typedef {import('./types.js').DeferredAttenuatorsProvider} DeferredAttenuatorsProvider */
+/** @typedef {import('./types.js').CompartmentDescriptor} CompartmentDescriptor */
+// get StaticModuleRecord from the ses package's types
+/** @typedef {import('ses').ModuleExportsNamespace} ModuleExportsNamespace */
 
-const { create, entries, values, assign, keys, freeze } = Object;
+const { entries, values, assign, keys, freeze } = Object;
+const { isArray } = Array;
 const q = JSON.stringify;
 
-const copyGlobals = (globals, list) => {
-  const copy = create(null);
-  if (list && list.length > 0) {
-    for (let index = 0; index < list.length; index += 1) {
-      const key = list[index];
-      // If an endowment is missing, global value is undefined.
-      // This is an expected behavior if globals are used for platform feature detection
-      copy[key] = globals[key];
-    }
-  }
-  return copy;
-};
-
 /**
  * Const string to identify the internal attenuators compartment
  */
 export const ATTENUATORS_COMPARTMENT = '<ATTENUATORS>';
 
+const ATTENUATOR_KEY = 'attenuate';
+const ATTENUATOR_PARAMS_KEY = 'params';
+const WILDCARD_POLICY_VALUE = 'any';
+const POLICY_FIELDS = ['builtins', 'globals', 'packages'];
+
+/**
+ *
+ * @param {unknown} item
+ * @param {string} path
+ * @param {string} [url]
+ * @param {object} [options]
+ * @returns {void}
+ */
+export const assertPolicyItem = (item, path, url, options = {}) => {
```

Comment:
I just realized this is not correct yet, we'll need separate validation logic for globals builtins and packages because they differ in how attenuation works.

Reply:
This may be related to a question I have: would it make more sense to have multiple single-purpose exports from the attenuator module?  Instead of just `attenuate`, we would have `attenuateGlobals` and `attenuateModule`.

I was thinking of bringing this up in our sync, but we ran out of time.

Reply:
It can work and it does help when defining a single default attenuator. On the other hand, it prevents me from using any simple attenuator implementation for whichever purpose I wish to use it. We could use attenute export as a default for both, but that feels like bikeshedding.